dali/test/dali_operator_test.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_TEST_DALI_OPERATOR_TEST_H_
+#define DALI_TEST_DALI_OPERATOR_TEST_H_
+
+#include <gtest/gtest.h>
+#include <vector>
+#include "tensor_adapter.h"
+#include "operators_graph.h"
+
+namespace dali {
+
+namespace testing {
+
+template<typename InputType, typename OutputType>
```

Comment:
I think it's too strong an assumption that operator has just one input type. You may, for example, want to apply a mask (probably uint8) to a floating point tensor.

Reply:
That's right. Would allowing `InputType` to be also a tuple be OK for you? Something like:

```
template<typename InputType, typename OutputType>
DaliOperatorTest : ::testing::Test {
  static_assert(is_tuple_of_fundamentals_or_fundamental<InputType>::value,"");
  static_assert(is_tuple_of_fundamentals_or_fundamental<OutputType>::value,"");
}
```

Reply:
Tuple is ok, although I don't know if we really need to make this "or" here. We might have some kind of selector that converts its arguments into a tuple, if necessary.
```
template <typename T>
using as_tuple = /* soething that converts T into a std::tuple<T> if T is not already a tuple */

template <typename input, typename output>
using OperatorTestBase = DaliOperatorTest<as_tuple<input>, as_tuple<output>>;
```